
Subject: [PATCH 08/12] ima: ima_delete_rules is now based on ima_free_policy_rules

ima_delete_rules() is now based on ima_free_policy_rules().
 
Signed-off-by: Guilherme Magalhaes <guilherme.magalhaes@hpe.com>
Signed-off-by: Joaquim Souza <joaquims@hpe.com>
Signed-off-by: Tycho Andersen <tycho@docker.com>
---
 security/integrity/ima/ima_policy.c | 11 +----------
 1 file changed, 1 insertion(+), 10 deletions(-)

diff --git a/security/integrity/ima/ima_policy.c b/security/integrity/ima/ima_policy.c
index f21a747..3c17c9f 100644
--- a/security/integrity/ima/ima_policy.c
+++ b/security/integrity/ima/ima_policy.c
@@ -891,17 +891,8 @@ void ima_free_policy_rules(struct list_head *policy_rules)
  */
 void ima_delete_rules(void)
 {
-	struct ima_rule_entry *entry, *tmp;
-	int i;
-
 	temp_ima_appraise = 0;
-	list_for_each_entry_safe(entry, tmp, &ima_temp_rules, list) {
-		for (i = 0; i < MAX_LSM_RULES; i++)
-			kfree(entry->lsm[i].args_p);
-
-		list_del(&entry->list);
-		kfree(entry);
-	}
+	ima_free_policy_rules(&ima_temp_rules);
 }
 
 #ifdef	CONFIG_IMA_READ_POLICY
-- 
2.7.4

